src/base/object_factory.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * object_factory.hpp
+ */
+
+#ifndef _OBJECT_FACTORY_HPP_
+#define _OBJECT_FACTORY_HPP_
+
+#ident "$Id$"
+
+#include <cassert>
+#include <functional>
+#include <map>
+
+/*
+ * factory of objects:
+ * main purpose is to be used with unpacking objects beloning to a class hierarchy:
+ * 'object_base' is the base class having derived classed 'object_type'
+ * each 'object_type' is identified by a value of 'object_key'
+ */
+namespace cubbase
+{
+  template <typename object_key, typename object_base> class factory
+    {
+      public:
+        template <typename object_type>
```

Comment:
usually there is no space between template and <...>
same thing for: `std::function <object_base*()> object_creator` 
consistency: sometimes you put a space, sometimes you don't
logic: in a declaration like `std::function <object_base*()> object_creator`  you should see 2 main parts (not 3): 
1. the type: `std::function<object_base*()>`
2. the variable: `object_creator`
see how it is usually used: http://en.cppreference.com/w/cpp/utility/functional/function